src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs

```diff
@@ -148,6 +148,12 @@ public class DownloadPipelineArtifactTaskV1 : PipelineArtifactTaskPluginBaseV1
             string tags = context.GetInput(ArtifactEventProperties.Tags, required: false);
             string userSpecifiedpipelineId = context.GetInput(PipelineId, required: false);
 
+            if (!Directory.Exists(targetPath) && !File.Exists(targetPath))
+            {
+                // if local path is neither file nor folder
+                throw new FileNotFoundException(StringUtil.Loc("PathNotExist", targetPath));
+            }   
```

Comment:
Why are changing V1 berhavior in a V2 PR?

Reply:
We had a discussion to auto-create directories if it doesn't exist, therefore removing this check in the V2 task

Reply:
Right by why are we changing behavior in V1 if the behavior change is supposed to be in V2?

Reply:
Removed